Adds the article: "Best 3 Workflow Orchestration Tools: a Comparison" (#1813)
kestra-io/docs: content/blogs/2024-10-DD-workflow-tools-comparison.md

---

> title: "Best 3 Workflow Orchestration Tools: a Comparison"

Reviewer: I'd word this specifically as data orchestration

```suggestion
title: "Data Orchestrators: How to Choose"
```

Reviewer: Same for the rest, reword this to be more focused on data

Reviewer: Use this to help: https://kestra.io/use-cases/data-engineers

Reviewer 2: this is an interesting point -- I think we could keep this article focused on workflow orchestration and have a separate one about data orchestrators;  in that case, Federico doesn't need to talk about Dagster here since the focus would be on orchestrating workflows, not data - WDYT @wrussell1999?

Reviewer: Agreed - let’s do this as a workflow orchestration one first and do data as a separate PR that includes Dagster. Cc @federico-trotta

Author: ![use-cases-data-engineers](https://github.com/user-attachments/assets/16b30c47-abcc-49f4-87a3-c032ef1243e7)
Just for the record: the page is broken... :(

Reviewer: Got an issue open to fix it, thanks

> description: "Discover the best 3 workflow orchestration tools, and why Kestra can be the best choice for your business.""

Reviewer: ```suggestion
description: "Discover the differences between the most popular data orchestrators, and how to decide which one is best for you."

```

Author: Applied, but removed "data"

date: 2024-10-xxxxx:00:00
category: xxxx
author:
  name: Federico Trotta
  image: "ftrotta"
image: /blogs/xxxxxxxxxxxxx.jpg
---

When it comes to automating and managing complex workflows, choosing the right workflow orchestration makes a massive difference for your company. As the choice of a workflow orchestration tool can simplify your processes or complicate them further - depending on how well it fits your needs -, making an informed decision is essential to ensure the success of your workflows and the overall efficiency of your team.

> But with so many orchestration tools available, how do you decide which one is the best fit for your needs?

Reviewer: I know we talked about replacing "orchestration tools" with "orchestrators" but I'd keep this one in

```suggestion
But with so many data orchestration tools available, how do you decide which one is the best fit for your needs?

```

Author: Not applied as we don't use "data" here.

In this article, we'll walk through what makes a workflow orchestration tool, how to pick one, and compare the top three market choices to help you make an informed decision that aligns with your needs, both present and future.

> ## What is a Workflow Orchestration Tool?

Reviewer: Redo this to focus on Data specifically, and data use cases like ETL

A workflow [orchestration tool](./2024-09-18-what-is-an-orchestrator.md) is software that automates the execution, monitoring, and management of tasks in a defined order across multiple systems or applications. It simplifies and schedules the complex processes behind the scenes, allowing engineers to focus on higher-level business logic rather than managing the infrastructure.

These tools are particularly useful for building and managing pipelines, automating business processes, and running batch operations. They ensure reliability by handling task retries in case of failures, provide fault tolerance to avoid disruptions, and deliver scalability to grow with your needs. This is especially important when dealing with large-scale distributed environments, where managing tasks manually would be impractical and error-prone.

> ## How to Choose a Workflow Orchestration Tool? 

Reviewer: Redo this to be more specific on data use cases

- **Extensibility**: At some point, you’ll  probably need a feature that isn’t supported out of the box. So, in this context, extensibility refers to how easily you can extend or customize the orchestration tool to meet your specific needs. This could be writing custom plugins, integrating with proprietary systems, or creating custom monitoring solutions, and a highly extensible tool offers hooks or APIs that allow you to build on top of it. 
- **Community & support**: Even the most powerful tools can feel like a dead end without a strong community or good support. An active user community means you’ll find plenty of resources — whether it’s documentation, tutorials, open-source plugins, or quick answers to your questions. Large communities are also a good sign that the tool is actively maintained and will continue evolving. Also, if your company is likely to need more personalized support, check if the tool offers a managed or enterprise version with dedicated support channels: this can be a critical aspect when you’re dealing with production issues and need timely resolutions.

So, with that in mind, let's compare three workflow orchestration tools.

> ## Top Three Workflow Orchestration Tools

Reviewer: I'd avoid doing this rating, and instead phrase it as a "frequently used" / "popular" tools

**Pros**:
- **Flexible execution model**: Prefect supports running workflows in a variety of environments — locally, in the cloud, or in a hybrid setup — without requiring changes to workflow code. This flexibility is ideal for organizations that need a mix of cloud and on-premises execution for different parts of their infrastructure.
- **Simplified infrastructure**: Compared to Airflow, Prefect's setup is more straightforward, reducing the initial overhead required to get started. This ease of deployment helps teams quickly get up and running, making Prefect a convenient choice for smaller projects or teams without dedicated infrastructure resources.

**Cons**:
- **Limited compared to DAGs**:  While simpler, Prefect's workflows lack some of the robustness and fine-grained control offered by Airflow's DAGs, particularly when dealing with highly complex data pipelines with multiple interdependencies.
- **Python-centric**: Prefect, like Airflow, is also Python-centric, which means teams that are not familiar with Python may find it challenging to use. 

> ## Why Kestra Stands Out

Reviewer: While I like this section, it makes it seem like the others don't have any of this. Look at this example for inspiration: https://kestra.io/blogs/data-orchestration-beyond-analytics

Author: I reworked this section, focusing on its unique characteristics (or, at least, the ones that are native)

- Iteratively build and refine workflows directly in the user interface, enabling rapid prototyping with immediate visual feedback.
- Test workflows in real-time, ensuring they perform as expected before moving them into production environments.
- Promote workflow code seamlessly to production, keeping it version-controlled and aligned with existing deployment practices without added friction.

By offering an integrated platform, Kestra reduces the barriers between different stages of the workflow lifecycle.

### Language-Agnostic
While Airflow and Prefect heavily rely on Python code to define workflows, Kestra offers a language-agnostic approach using YAML. This allows teams without programming expertise to create and manage workflows through declarative configurations, enhancing collaboration between technical and non-technical stakeholders.

This approach simplifies maintenance and reduce the likelihood of errors, as workflows are defined through straightforward YAML configurations that are easier to read and write. 

> Additionally, Kestra's YAML configurations enable seamless integration with different tech stacks like [Go](../docs/15.how-to-guides/golang.md) and [Rust](../docs/15.how-to-guides/rust.md). This means that engineers can write custom tasks and plugins in their preferred languages and incorporate them directly into their workflows. This flexibility allows teams to leverage existing codebases and expertise across various programming languages, fitting seamlessly into diverse engineering environments and enabling the use of different tools for specific needs.

Reviewer: ```suggestion
Additionally, Kestra's declarative syntax with YAML seamlessly integrates with different tech stacks like [Python](../docs/15.how-to-guides/python.md), [Go](../docs/15.how-to-guides/golang.md) and [Rust](../docs/15.how-to-guides/rust.md). This means that engineers can write custom tasks and plugins in their preferred languages and incorporate them directly into their workflows. This flexibility allows teams to leverage existing codebases and expertise across various programming languages, fitting seamlessly into diverse engineering environments and enabling the use of different tools for specific needs.
```

These tools are particularly useful for building and managing pipelines, automating business processes, and running batch operations. They ensure reliability by handling task retries in case of failures, provide fault tolerance to avoid disruptions, and deliver scalability to grow with your needs. This is especially important when dealing with large-scale distributed environments, where managing tasks manually would be impractical and error-prone.

## How to Choose a Workflow Orchestration Tool? 
When choosing the best workflow orchestration tool for your needs, it’s important to think beyond just the immediate needs of your project. The tool you're choosing, in fact, should not only be a good fit for today's needs: it also must be able to scale with your future workflows.

So, here’s a look at the key factors to consider when making your choiche:

> - **Ease of use**: How easy is it to get started with the tool? This doesn’t just mean a smooth installation: it extends to the entire user experience. A good workflow orchestration tool, in fact, should have an intuitive interface, so that managing and monitoring workflows doesn’t feel like rocket science. 

Reviewer: ```suggestion
- **Ease of use**: How easy is it to get started with the tool? This doesn’t just mean a smooth installation: it extends to the entire user experience. A good workflow orchestration tool should have an intuitive interface, so that managing and monitoring workflows doesn’t feel like rocket science. 

```

- **Community & support**: Even the most powerful tools can feel like a dead end without a strong community or good support. An active user community means you’ll find plenty of resources — whether it’s documentation, tutorials, open-source plugins, or quick answers to your questions. Large communities are also a good sign that the tool is actively maintained and will continue evolving. Also, if your company is likely to need more personalized support, check if the tool offers a managed or enterprise version with dedicated support channels: this can be a critical aspect when you’re dealing with production issues and need timely resolutions.

So, with that in mind, let's compare three workflow orchestration tools.

## Comparing Popular Workflow Orchestration Tools

> Let’s dive into three of the most used workflow orchestration tools: Kestra, Apache Airflow, and Prefect. Each has its strengths and weaknesses, but one stands out for its flexibility, ease of use, and innovation.

Reviewer: Could you reword this to focus on how each one suits specific use cases and do those well? Kestra on the whole does stuff well which goes into the whole scalability and flexibility point.

```suggestion
Let’s dive into three of the most used workflow orchestration tools: Kestra, Apache Airflow, and Prefect. Each has its strengths and weaknesses, but one stands out for its flexibility, ease of use, and innovation.

```

### Kestra
![Kestra UI](/blogs/2024-10-DD-workflow-tools-comparison/kestra-UI.png)

[Kestra](https://kestra.io/) is a modern, distributed orchestration tool built for handling complex workflows at scale. It offers flexibility and modularity, making it adaptable to various workflows, from basic task automation to managing massive data pipelines. With a focus on scalability and high availability, Kestra is optimized for modern cloud environments and is designed with a cloud-native architecture, making it a great fit for modern systems that need to scale seamlessly as demands grow.

> **Unique feature: language-agnostic workflows**

Reviewer: also another key thing that makes us unique is being able to build workflows as code (YAML) or directly from the UI bridging the gap between technical users and non technical users.

- **Flexible database backends**: One of Kestra’s standout features is its support for multiple database backends. This flexibility allows users to choose the backend that best fits their performance, scalability, and integration needs
- **Modern design**:  Built with cloud-native architectures in mind, Kestra embraces modern design principles, such as microservices and distributed computing, to optimize performance in dynamic environments. Its architecture supports horizontal scaling across multiple nodes, making it easy to add capacity as workflows grow. 
- **Intuitive UI**: Kestra’s user interface  is designed with simplicity and usability in mind, providing an intuitive visual representation of workflows. The UI makes it easy to create, configure, and monitor workflows, even for users who don’t have extensive technical expertise.

**Cons**:

> - **Smaller community**: Being newer on the market, Kestra may have a smaller user base with respect to other tools. Though, is rapidly growing.

Reviewer: ```suggestion
- **Smaller community**: Being newer on the market, Kestra may have a smaller user base with respect to other tools. Though, is rapidly growing and very active.

```

- Promote workflow code seamlessly to production, keeping it version-controlled and aligned with existing deployment practices without added friction.

In contrast, Airflow and Prefect often require fragmented processes for development, testing, and deployment, which can introduce complexity and slow down the workflow lifecycle:

- Airflow requires workflows to be defined in Python, which is done outside of its web-based user interface. This separation means that developers often use different tools for coding, testing, and deploying workflows, which can complicate collaboration and alignment between development and production environments.
- Prefect also relies on Python scripts for workflow definitions, and while it offers a UI for monitoring, the development and testing phases are primarily code-centric and may not provide the same level of integration between iterative development and deployment. So, promoting workflows to production can require additional configurations.

So, by offering an integrated platform, Kestra reduces the barriers between different stages of the workflow lifecycle.

### Language-Agnostic and Code-Free Workflow Definitions

> While Airflow and Prefect heavily rely on Python code to define workflows, Kestra offers a language-agnostic, code-free approach using YAML. This allows teams without programming expertise to create and manage workflows through declarative configurations, enhancing collaboration between technical and non-technical stakeholders.

Reviewer: Code free might be a bad thing to some people :D 

```suggestion
While Airflow and Prefect heavily rely on Python code to define workflows, Kestra offers a language-agnostic approach using YAML. This allows teams without programming expertise to create and manage workflows through declarative configurations, enhancing collaboration between technical and non-technical stakeholders.

```

So, with its distributed design, Kestra provides seamless scalability out of the box, making it well-suited for enterprises with demanding workloads and the need for reliable, high-performance orchestration.

### Advanced Error Handling and Fault Tolerance
Kestra provides sophisticated error handling mechanisms and fault tolerance features that enhance the reliability of workflows. It supports complex retry strategies, conditional execution paths, and dynamic handling of failures to ensure that workflows can recover gracefully from errors.

On the contrary:

- Airflow offers basic retry and error handling but may require custom code to implement more advanced fault tolerance strategies. SO, managing complex error scenarios can become cumbersome and increase code complexity.
- Prefect provides some error handling features but may not match the depth of Kestra's built-in capabilities for managing intricate failure scenarios.

So, with this feature, Kestra reduces the need for custom error-handling code, simplifies workflow definitions, and improves the overall system robustness.

> ### Native Cloud FeaturesS

Reviewer: ```suggestion
### Native Cloud Features

```

Kestra is designed with cloud-native environments in mind, enabling efficient execution of containerized tasks and leveraging cloud infrastructure for scaling and resource management.

Instead:

- Airflow can be deployed in cloud environments but wasn't originally designed with cloud-native principles, which may result in additional configuration and management overhead for seamless integration. 

> - Prefect can also run in the cloud but lacks the same level of native integration with container orchestration platforms like Kubernetes.

Reviewer: Looks like Prefect does work with K8s for flows? https://docs.prefect.io/3.0/deploy/infrastructure-examples/kubernetes

Author: Also Airflow does (https://airflow.apache.org/docs/apache-airflow/stable/administration-and-deployment/kubernetes.html). My understanding is that Kestra makes it straightforward with respect to them because its architecture is cloud-native (while the others are not).

Anyway, as it may lead to misleading (i.e.: Kestra works with k8s while the others don't), I have rephrased it to be more "general and high-level", saying that Kestras' cloud-native features simplify deployment.  For the other two, instead, there can be complexities during setup and management (because their architecture is not cloud-native).

But, indeed: we can remove this paragraph if you think it's better to do so. :)

- **Kestra vs Prefect**: Choose Kestra when scalability and flexibility are critical, such as in enterprise environments with complex workflows. Prefect’s hybrid model is valuable for balancing security with cloud orchestration, but Kestra's architecture ensures better scalability for demanding workflows. Finally, Kestra’s language-agnostic approach, intuitive UI, and customization options make it a future-proof solution that can handle a wide variety of orchestration needs.

## Conclusions
When it comes to choosing the right workflow orchestration tool, there’s no one-size-fits-all solution: the best tool depends on your specific needs, including the complexity of your workflows, scalability requirements, and your team's expertise.

> Kestra, however, stands out as the strongest option for modern, cloud-native environments. Unlike Airflow - which can struggle with scaling and has a steeper learning curve -, or Prefect - which may lack some robustness in complex workflows -, Kestra strikes the perfect balance between accessibility and power. Its language-agnostic approach and intuitive UI, in fact, make it accessible for teams of all sizes and skill levels, while its highly scalable architecture ensures that it can handle even the most demanding workflows, allowing you to grow without worring about managing infrastructure.

Reviewer: ```suggestion
Kestra, however, stands out as the strongest option for modern, cloud-native environments. Unlike Airflow, which can struggle with scaling and has a steeper learning curve, or Prefect, which may lack some robustness in complex workflows; Kestra strikes the perfect balance between accessibility and power. Its language-agnostic approach and intuitive UI, in fact, make it accessible for teams of all sizes and skill levels, while its highly scalable architecture ensures that it can handle even the most demanding workflows, allowing you to grow without worring about managing infrastructure.
```

Let's now describe each tool, by providing an overview, a unique feature, and pros and cons to help you get a better understanding for your choiche.

### Kestra
![Kestra UI](/blogs/2024-10-DD-workflow-tools-comparison/kestra-UI.png)

[Kestra](https://kestra.io/) is a modern, distributed orchestration tool built for handling complex workflows at scale. With a focus on scalability and high availability, Kestra is optimized for modern cloud environments and is designed with a cloud-native architecture, making it a great fit for systems that need to scale seamlessly as demands grow. Moreover, it enables users to build workflows either by coding in YAML or directly through its intuitive user interface, bridging the gap between technical and non-technical users and fostering collaboration across different teams.

**Unique feature: language-agnostic workflows**

> Kestra provides YAML-based workflow definitions which make it a language-agnostic tool. This means that teams can use Kestra regardless of their preferred programming language, making it easy to adopt across different departments or tech stacks. This flexibility allows for greater collaboration across teams and makes it an excellent fit for organizations that want a standardized workflow tool without needing programming skills focused on a particular language.

Author: Hope I understood this correctly...

[Apache Airflow](https://airflow.apache.org/) is one of the most well-known and widely used workflow orchestration tools. Built by Airbnb and now part of the Apache Software Foundation, Airflow excels at managing workflows in Python with a strong focus on ETL pipelines. It also provides a highly customizable solution for engineers familiar with Python and is widely used in the data engineering and analytics domains.

**Unique feature: DAG-based workflow management**

Airflow uses Directed Acyclic Graphs (DAGs) to define task dependencies, which is useful for data pipelines requiring specific execution sequences. DAGs, in fact, allow users to visualize the entire workflow structure, understand task dependencies, and control the execution order with precision: this feature is important for workflows focused on data engineering, where task order is crucial for processing data accurately. 

**Pros**:
- **Large community and ecosystem**: With its long-standing presence, Airflow has amassed a large community and an extensive ecosystem. Numerous plugins and resources are available, making it easy to find solutions, tutorials, and pre-built connectors for common use cases.

> - **Customizability**: Airflow offers full flexibility to define workflows using Python, making it a preferred choicge among Python developers. 

Reviewer: ```suggestion
- **Customizability**: Airflow offers full flexibility to define workflows using Python, making it a preferred choice among Python developers. 
```

**Cons**:
- **Scaling challenges**: Airflow's dependence on a centralized database can lead to performance bottlenecks as workloads grow. This can create a single point of failure, and as workflows increase in number and complexity, scaling becomes challenging

> - **Python-based**: Airflow’s reliance on Python means that non-Python developers might face a steep learning curve, limiting its accessibility for mixed-skill teams.

Reviewer: ```suggestion
- **Python-based**: Airflow’s reliance on Python means that non-Python developers might face a steep learning curve, limiting its accessibility for mixed-skill teams. On the counter, this might be a positive if your tech stack is heavily based on Python.

```

- **Older architecture**: Airflow was not designed with cloud-native principles in mind, so its older architecture may require additional effort to adapt to modern cloud-native use cases.

### Prefect

![Prefect UI](/blogs/2024-10-DD-workflow-tools-comparison/prefect-UI.png)

[Prefect](https://www.prefect.io/) is an open-source orchestration tool designed to simplify the process of building and monitoring data workflows. Prefect offers both self-hosted and cloud-based solutions, providing flexibility in deployment. Workflows in Prefect are defined using Python, which makes it easy for data scientists and engineers to integrate them into existing Python-based projects.

> **Unique feature: hybrid execution model**

Reviewer: I'm not sure what this means and if it will be clear to users. Maybe revisit this page for inspiration: https://kestra.io/vs/prefect

Reviewer: @anna-geller might have some good suggestions for the Prefect section too

Author: I reprhased removing the "simplify [...]" part that, indeed, is not objective (what's simple for me can be hard for someone else :) ). 

Let me know if this is sufficient... :)

**Pros**:
- **Flexible execution model**: Prefect supports running workflows in a variety of environments — locally, in the cloud, or in a hybrid setup — without requiring changes to workflow code. This flexibility is ideal for organizations that need a mix of cloud and on-premises execution for different parts of their infrastructure.
- **Simplified infrastructure**: Compared to Airflow, Prefect's setup is more straightforward, reducing the initial overhead required to get started. This ease of deployment helps teams quickly get up and running, making Prefect a convenient choice for smaller projects or teams without dedicated infrastructure resources.

**Cons**:
- **Limited compared to DAGs**:  While simpler, Prefect's workflows lack some of the robustness and fine-grained control offered by Airflow's DAGs, particularly when dealing with highly complex data pipelines with multiple interdependencies.

> - **Python-centric**: Prefect, like Airflow, is also Python-centric, which means teams that are not familiar with Python may find it challenging to use. 

Reviewer: ```suggestion
- **Python-centric**: Prefect, like Airflow, is also Python-centric, which means teams that are not familiar with Python may find it challenging to use. Again, this might be a benefit if your tech stack is primarily Python based.
```

## Why Kestra Stands Out

> While Apache Airflow and Prefect are powerful workflow orchestration tools, Kestra introduces unique features and capabilities that distinguish it from the rest. Let's explore the aspects where Kestra excels beyond what Airflow and Prefect offer.

Reviewer: I don't think this adds much - would prefer if we just jump straight into the answers
```suggestion
```

- Promote workflow code seamlessly to production, keeping it version-controlled and aligned with existing deployment practices without added friction.

In contrast, Airflow and Prefect often require fragmented processes for development, testing, and deployment, which can introduce complexity and slow down the workflow lifecycle:

- Airflow requires workflows to be defined in Python, which is done outside of its web-based user interface. This separation means that developers often use different tools for coding, testing, and deploying workflows, which can complicate collaboration and alignment between development and production environments.
- Prefect also relies on Python scripts for workflow definitions, and while it offers a UI for monitoring, the development and testing phases are primarily code-centric and may not provide the same level of integration between iterative development and deployment. So, promoting workflows to production can require additional configurations.

So, by offering an integrated platform, Kestra reduces the barriers between different stages of the workflow lifecycle.

### Language-Agnostic and Code-Free Workflow Definitions

> While Airflow and Prefect heavily rely on Python code to define workflows, Kestra offers a language-agnostic approach using YAML. This allows teams without programming expertise to create and manage workflows through declarative configurations, enhancing collaboration between technical and non-technical stakeholders.

Reviewer: Also worth noting that you can use different tech stacks too, like Go or Rust

Author: Not sure if the "solution" I provided fits your expectations about that...let me know and, eventually, we'll modify it accordingly :)

On the other hand:

- Airflow can face scaling challenges due to its centralized scheduler and reliance on a single database, which can become bottlenecks as workloads grow. So, scaling Airflow often requires complex configurations and may introduce single points of failure.
- Prefect offers scalability but may require additional setup and infrastructure considerations to achieve distributed execution, as it doesn't natively leverage technologies like Kafka for high-throughput scenarios.

So, with its distributed design, Kestra provides seamless scalability out of the box, making it well-suited for enterprises with demanding workloads and the need for reliable, high-performance orchestration.

### Advanced Error Handling and Fault Tolerance
Kestra provides sophisticated error handling mechanisms and fault tolerance features that enhance the reliability of workflows. It supports complex retry strategies, conditional execution paths, and dynamic handling of failures to ensure that workflows can recover gracefully from errors.

On the contrary:

> - Airflow offers basic retry and error handling but may require custom code to implement more advanced fault tolerance strategies. SO, managing complex error scenarios can become cumbersome and increase code complexity.

Reviewer: These comparisons feel very harsh to the other tools. It would be beneficial to elaborate otherwise people might stop reading as the comparisons are starting to look quite bias 😅 

```suggestion
- Airflow offers basic retry and error handling but may require custom code to implement more advanced fault tolerance strategies. Managing complex error scenarios can become cumbersome and increase code complexity.

```

Author: Yeah: re-reading it after some time has passed I definitely agree. I removed all the comparisons (a part from the one on cloud-native features which is "smooth", in my opinion), and made this paragraph Kestra-centered. Let me know if this may fit... :)

- Prefect provides some error handling features but may not match the depth of Kestra's built-in capabilities for managing intricate failure scenarios.

So, with this feature, Kestra reduces the need for custom error-handling code, simplifies workflow definitions, and improves the overall system robustness.

### Native Cloud Features

> Kestra is built from the ground up with [cloud-native](docs/blob/main/pages/use-cases/modern-data-stack.vue) principles in mind, making it a natural fit for modern cloud environments. Its architecture seamlessly leverages containerization and orchestration technologies, which means it can scale efficiently, manage resources effectively, and integrate smoothly with cloud infrastructures.

Reviewer: ```suggestion
Kestra is built from the ground up with [cloud-native](/use-cases/modern-data-stack) principles in mind, making it a natural fit for modern cloud environments. Its architecture seamlessly leverages containerization and orchestration technologies, which means it can scale efficiently, manage resources effectively, and integrate smoothly with cloud infrastructures.

```

On the other hand, while both Prefect and Airflow have made strides in supporting cloud deployments, they were not originally built with a cloud-native mindset, and this can lead to complexities during setup and management.

> ## When to Choose Kestra Over Airflow and Prefect

Reviewer: I think it would be good to make this more balanced.

Maybe something like

Kestra for X reason

Airflow for Y reason

Prefect for Z reason

Helps makes us look less bias while showing how Kestra stands out
